### PR TITLE
Featured posts on first page

### PR DESCRIPTION
**Pull request by b-m-f**
# Done
- Fetches 3 featured posts on the first page of the blog.

# QA
Run the tests with `poetry install && poetry run pytest`

### CODE REVIEW

File: canonicalwebteam/blog/django/views.py [hunk 1]
```diff
@@ -15,16 +15,39 @@
...
+            articles, total_pages = api.get_articles(
+                tags=tag_ids, page=page_param
+            )
+            featured_articles = []
+
     except Exception as e:
         return HttpResponse("Error: " + e, status=502)
 
-    context = get_index_context(page_param, articles, total_pages)
+    print(get_index_context)
```
Comment (nottrobin) [inline canonicalwebteam/blog/django/views.py]: Did you accidentally leave this print statement in?

File: canonicalwebteam/blog/django/views.py [hunk 2]
```diff
@@ -15,16 +15,39 @@
...
     try:
-        articles, total_pages = api.get_articles(
-            tags=tag_ids, tags_exclude=excluded_tags, page=page_param
-        )
+        if page_param == "1":
+            featured_articles, total_pages = api.get_articles(
+                tags=tag_ids,
+                tags_exclude=excluded_tags,
+                page=page_param,
+                sticky="true",
```
Comment (nottrobin) [inline canonicalwebteam/blog/django/views.py]: I assume this is what makes them "featured"?

File: canonicalwebteam/blog/common_view_logic.py [hunk 3]
```diff
@@ -5,15 +5,15 @@
...
+def get_index_context(page_param, articles, total_pages, featured_articles=[]):
     """
     Build the content for the index page
     :param page_param: String or int for index of the page to get
     :param articles: Array of articles
     :param articles: String of int of total amount of pages
     """
 
-    for article in articles:
+    def transform_index_article(article):
```
Comment (nottrobin) [inline canonicalwebteam/blog/common_view_logic.py]: Why  is this nested inside `get_index_context`? I don't think it needs to be does it? It's only needs the article that's passed to it right?

If you've no objection, could you please move this function to being top level? (And ideally give it a docstring)

File: canonicalwebteam/blog/django/views.py (hunk 1 above)
Comment (b-m-f) [reply canonicalwebteam/blog/django/views.py]: oh, thanks :)

File: canonicalwebteam/blog/django/views.py (hunk 2 above)
Comment (b-m-f) [reply canonicalwebteam/blog/django/views.py]: Yes, exactly.

**Comment by nottrobin** [comment]
I get:

``` bash
$ poetry run pytest
Creating virtualenv canonicalwebteam.blog-py3.6 in /home/robin/.cache/pypoetry/virtualenvs
                                                           
[FileNotFoundError]                         
[Errno 2] No such file or directory: 'pytest': 'pytest'  
                                                           
run <args> (<args>)...
```

Is that expected? Is that what you meant when you said you were giving up on tests?

**Pull request merged**